javaparser-core/src/main/java/com/github/javaparser/utils/TypeUtils.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2007-2010 Júlio Vilmar Gesser.
+ * Copyright (C) 2011, 2013-2022 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+package com.github.javaparser.utils;
+
+import java.lang.reflect.Method;
+
+import com.github.javaparser.ast.type.PrimitiveType;
+import com.github.javaparser.ast.type.VoidType;
+
+public class TypeUtils {
+
+    /*
+     * Returns the method descriptor (https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.3.3)
+     * The method descriptor for the method: {@code Object m(int i, double d, Thread t) {...}}
+     * is {@code (IDLjava/lang/Thread;)Ljava/lang/Object;}
+     * Note that the internal forms of the binary names of Thread and Object are used.
+     */
+    public static String getMethodDescriptor(Method method) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("(");
+        for (Class<?> parameter : method.getParameterTypes()) {
+            appendDescriptor(parameter, stringBuilder);
+        }
+        stringBuilder.append(")");
+        appendDescriptor(method.getReturnType(), stringBuilder);
+        return stringBuilder.toString();
+    }
+
+    private static void appendDescriptor(final Class<?> clazz, final StringBuilder stringBuilder) {
+        Class<?> currentClass = clazz;
+        while (currentClass.isArray()) {
+            stringBuilder.append("[");
+            currentClass = currentClass.getComponentType();
+        }
+        if (currentClass.isPrimitive()) {
+            String descriptor = getPrimitiveTypeDescriptor(currentClass);
+            stringBuilder.append(descriptor);
+        } else {
+            stringBuilder.append("L").append(currentClass.getName().replace(".", "/")).append(";");
+        }
+    }
+
+    public static String getPrimitiveTypeDescriptor(final Class<?> clazz) {
+        String descriptor;
+        if (clazz == Void.TYPE) {
+            descriptor = new VoidType().toDescriptor();
+        } else if (clazz == Integer.TYPE) {
```

Comment:
Maybe you can simplify this structure by using PrimitiveType.byBoxTypeQName(..)

Reply:
I think such a method only exists for ResolvedPrimitiveType and not PrimitiveType. Unfortunately, `ResolvedPrimitiveType.byBoxTypeQName` won't be useful here as it only returns Resolved(Primitive)Types.

But I can implement a similar method for PrimitiveTypes. Maybe like this:
```java
public static Optional<Primitive> getPrimitiveByBoxedTypeName(String name) {
    for (Primitive prim : Primitive.values()) {
        if (prim.nameOfBoxedType.equals(name)) {
            return Optional.of(prim);
        }
    }
    return Optional.empty();
}
```

And then in `TypeUtils.getPrimitiveTypeDescriptor` I can do:
```java
// void case:
...
// primitve cases:
Optional<Primitive> optPrim = PrimitiveType.getPrimitiveByBoxedTypeName(clazz.getCanonicalName());
if (optPrim.isPresent()) {
    return new PrimitiveType(optPrim.get()).toDescriptor();
}
```

Reply:
> I think such a method only exists for ResolvedPrimitiveType and not PrimitiveType. Unfortunately, `ResolvedPrimitiveType.byBoxTypeQName` won't be useful here as it only returns Resolved(Primitive)Types.

You are right I meant ResolvedPrimitiveType.
Why do you say it's not useful? You can do something like this (or something better with a little more work).

    @Test
	void test() {
		Optional<ResolvedType> rt = ResolvedPrimitiveType.byBoxTypeQName(Integer.class.getCanonicalName());
		String type = toDescritor(rt.get().asPrimitive().name());
		assertEquals("I", type);
	}
	
	private String toDescritor(String primitiveTypeName) {
		return getPrimitiveType(primitiveTypeName)
				.map(p -> new PrimitiveType(p).toDescriptor())
				.orElseThrow(() -> new IllegalArgumentException("Unknown primitive"));
	}
	
	private Optional<Primitive> getPrimitiveType(String name) {
		return Optional.of(Primitive.valueOf(name));
	}

Reply:
Ah, that's clever.
What are your thoughts on adding a (getter) method `toDescriptor` to `Primitive` and then
```java
public String getPrimitiveTypeDescriptor(final Class<?> clazz) {
    String className = clazz.getCanonicalName();
    Optional<Primitive> optPrim = Optional.of(Primitive.valueOf(className));
    return optPrim.map(p -> p.toDescriptor())
            .orElseThrow(() -> new IllegalArgumentException("Unknown primitive"));
}
```

Reply:
Why not but in this case it is also necessary to adapt the toDescriptor method in the PrimitiveType class.

Reply:
The new commit should contain the mentioned improvements. Additionally I realized, that before only for the 'real' primitive classes (int.class, byte.class etc.) the correct descriptor was returned. Passing a wrapper class (like Integer.class) into the method resulted in an error.

I fixed that. Now you should get the correct descriptor for all primitive classes as well as their wrapper classes.
The solution looks like this:
```java
public static String getPrimitiveTypeDescriptor(final Class<?> clazz) {
    if (clazz == Void.TYPE || clazz == Void.class) {
        return new VoidType().toDescriptor();
    }
    String className = clazz.getCanonicalName();
    Optional<Primitive> prim = getPrimitive(className);
    if (prim.isPresent()) {
        return prim.get().toDescriptor();
    }
    Optional<ResolvedType> rt = ResolvedPrimitiveType.byBoxTypeQName(className);
    if (rt.isPresent()) {
        return getResolvedPrimitiveTypeDescriptor(rt);
    }
    throw new IllegalArgumentException("Unknown primitive: " + className);
}

private static String getResolvedPrimitiveTypeDescriptor(Optional<ResolvedType> rt) throws AssertionError {
    String typeName = rt.get().asPrimitive().describe();
    Optional<Primitive> prim = getPrimitive(typeName);
    return prim.map(p -> p.toDescriptor())
            .orElseThrow(() -> new AssertionError(
                    String.format(
                            "ResolvedPrimitiveType name \"%s\" does not match any Primitive enum constant identifier.",
                            typeName.toUpperCase())));
}

private static Optional<Primitive> getPrimitive(String name) {
    try {
        return Optional.of(Primitive.valueOf(name.toUpperCase()));
    } catch (IllegalArgumentException e) {
        return Optional.empty();
    }
}
```

But now I am honestly wondering, if it would'nt  be better to just do:
```java
if (clazz == Void.TYPE || clazz == Void.class) {
    descriptor = new VoidType().toDescriptor();
} else if (clazz == Integer.TYPE || clazz == Integer.class) {
    descriptor = Primitive.INT.toDescriptor();
} else if (clazz == Boolean.TYPE || clazz == Boolean.class) {
    ...
```